### DISCUSSION TITLE
What should a screening API filter look like?

### DISCUSSION THREAD

**Discussion by hburn7** [osu-tournament-rating/otr-api#209] [Q&A]
There are plans for us to provide a screening API to hosts/developers that are interested in using o!TR for their tournaments. This would be a service that lets hosts feed in a list of players to our API and get a response with valid/invalid players based on some customizable criteria.

This discussion is focused on the following question: **What criteria should we allow hosts to screen for?**

We have come up with the following so far, but would like to hear from others on what we can include. Please keep responses specific to o!TR. Note, **all of the below criteria can be combined or used by itself as a screening mechanism.** Keep in mind, below is only based on the tournament data we have in our database.

Planned:
- Ruleset (osu!, osu!mania, etc.)
- Min rating
- Max rating
- Provisional status (provisional = the algorithm is not confident in their current rating, or user does not have enough tournaments or matches played)
- Number of tournaments played
- Number of matches played

Suggestions:
- Rank range (allows screening based on # of tournaments played at this rank range or worse)
- Time since last match
- Peak rating

**Comment by cytusine0** [comment]
Since we already collect information on tournaments during submission, hosts may find criteria like "number of tournaments played with upper rank range 10k or worse" useful. Same idea with team sizes -- it feels a little weird right now to screen by "number of 1v1 10k tournaments played," but it seems doable and possibly useful in some narrow circumstance.

Also just a note to be careful about separating out tournaments of different gamemodes (so we'll want to consider tournament-level data regardless)!

**Reply by hburn7** [reply]
Sure, I think the logic would be like `?rankRange=10000&tournamentsPlayed=5` for "player must have played 5 tournaments of 10k+ skill level". Perhaps querying for rankRange by itself (without tournaments played) would screen anyone who has never played a tournament of this skill range.

**Comment by Userlog2** [comment]
Some ideas:
- The ability to screen based on where someone has placed in a tournament (or tournaments) - a criteria like "has placed top 3 in a 150k tournament" or "has placed top 8 in 2 100k tournaments" seems fairly useful as a supplementary or alternative screening method. If the specific placement is not available (e.g. place 7 in single-elimination), it should be generous and use the nearest one available that is less strict. (Hopefully, the host knows what specific placements are available and avoids this, but just in case...)
- If "archived" tournaments (tournaments that would be accepted, but break criteria) are stored in o!TR, an override that says that archived tournaments should be included in the screening could be useful; for example, if a tournament wants to screen based on past performance in 6-digit regionals or gimmick tournaments, it could be useful.
- The option to filter by date of the tournament's ending. For example, a tournament may want to screen only based on players' performances on tournaments within the last 12 months.

**Reply by hburn7** [reply]
1. Unfortunately, no way for us to track placements correctly. This goes beyond the scope of o!TR.
2. I'd need some feedback from Cytusine on this but my gut reaction is to not allow this as it kind of defeats the purpose, i.e. why screen from data that is not actually in the rating system?
3. This could definitely be added, good idea.

**Reply by Userlog2** [reply]
I intend overrides to be allowed only in limited tournaments (unbadged ~150k-250k), and should be avoided in serious/badged tournaments.

**Reply by cytusine0** [reply]
1. Yeah, tracking placements is unfortunately difficult because of forfeits and delays in scheduling, and manually matching each match in the database with a specific spot in a Challonge bracket would be too much of a task to take on. The best we can do is that every match is linked to the forum post from which the tournament came, so it's not too much work to find out who did actually win a particular tournament.
3. Basically agreed with Stage. We're screening those tournaments out not solely because they're "not near badging criteria" but also because there isn't enough interaction among the rank range there to get meaningful numbers, so we'd like to not try to measure performance in those tournaments. But it does depend a little on what we're thinking about in the long run with rejected tournaments -- are we still displaying the match data on a tournaments page but just not counting it for any rating changes?

**Comment by SourMongoose** [comment]
Current rating?

**Reply by hburn7** [reply]
Would be covered by
> Min rating
> Max rating

**Reply by SourMongoose** [reply]
Not sure how current rating is covered by those two

As an example, if someone started at 400, went up to 2000 a few years ago, and is now sitting at 1600, I'd want to screen them based on their 1600 rating

**Reply by hburn7** [reply]
Sorry, the way it's worded can cause some confusion.

Min rating = "Filter players that have a current rating below this minimum value."
Max rating = "Filter players that have a current rating above this maximum value."

**Reply by SourMongoose** [reply]
I see, then I'd also add a filter for peak rating
